### PR TITLE
Updates for 0.2.2 release

### PR DESCRIPTION
**Pull request by jtniehof**
This PR makes all the changes relevant to doing the 0.2.2 release. It is probably going to be easier to review by reading the commit history instead of just the overall diff. The changes include:

- Update documentation on the Python 2 transition and related (removing less-used Windows binaries)
- General installation documentation updates. These probably could use a full scrub/rewrite again soon, as they're growing in complexity again, but I don't think it needs to be a blocker right now. At the same time, suggestions for improvements still welcome.
- Update build scripts and support scripts for the current situation (3.9 exists but not in conda, Windows bug affecting numpy)
- New leapsecond file (no changes, but it has a current timestamp) because we now have a warning for potentailly out-of-date leapsecond file
- Change all versions to 0.2.2. This is the commit that will be tagged as the release.
- Change all versions to 0.2.3pre. The actual release will happen from the commit _prior_ to this (and will be tagged), but it seemed to make more sense to get that commit in here so it's all in a piece, rather than having to do another PR just for these changes. It also avoids the problem of having other PRs potentially merged in between the 0.2.2 release one and setting numbers to 0.2.3pre.

## PR Checklist
- [X] Pull request has descriptive title
- [X] Pull request gives overview of changes
- [X] New code has inline comments where necessary
- [X] (N/A) Any new modules, functions or classes have docstrings consistent with SpacePy style
- [X] Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [X] (N/A) New features and bug fixes should have unit tests
- [X] (N/A) Relevant issues are linked to (e.g. `See issue #` or `Closes #`)

### CODE REVIEW

**Comment by jtniehof** [comment]
I have the release scheduled for Monday assuming there aren't any blockers.

File: developer/scripts/win_build_system_teardown.cmd [hunk 1]
```diff
@@ -7,4 +7,8 @@
 %USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py36_64
 %USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py37_32
 %USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py37_64
+%USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py38_32
+%USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py38_64
+%USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py39_32
+%USERPROFILE%\Miniconda3\Scripts\conda env remove -y --name py39_64
```
Comment (balarsen) [inline developer/scripts/win_build_system_teardown.cmd:13]: Per https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#removing-an-environment I think you want the `--all` here too. Not likely critical but does a full cleanup.

**Review by balarsen — COMMENTED** [review]
Looks good, one comment/question on `developer/scripts/win_build_system_teardown.cmd`

File: developer/scripts/win_build_system_teardown.cmd (hunk 1 above)
Comment (jtniehof) [reply developer/scripts/win_build_system_teardown.cmd:13]: That link says either `remove --name NAME --all` or `env remove --name NAME`. This is using the `env remove` syntax. No indication in there of a `env remove --all` possibility.

**Review by spacecataz — DISMISSED** [review]
Thanks look good for me.  If there's nothing else holding us back, I can approve.

**Review by drsteve — APPROVED** [review]

**Review by spacecataz — APPROVED** [review]

**Pull request merged**